libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -808,7 +808,16 @@ void AP_GPS::update_instance(uint8_t instance)
             state[instance].uart_timestamp_ms = 0;
         }
         // delta will only be correct after parsing two messages
-        timing[instance].delta_time_ms = tnow - timing[instance].last_message_time_ms;
+        // Make sure that the delta_time is not negative if time is not monotonic
+        if(tnow >= timing[instance].last_message_time_ms)
```

Comment:
In general we can't directly compare times like this because it will cause problems when the time wraps back to zero.

A small formatting fix would be to put the brackets on the line before.  So for example it should be "} else {"